From e88357c5f210a8796104505c090fb6a04c213902 Mon Sep 17 00:00:00 2001
From: Philip Chimento <philip.chimento@gmail.com>
Date: Mon, 21 Nov 2011 22:26:21 +0100
Subject: [PATCH 1/2] Change IgeMacIntegration to GtkOSXApplication

Instead of using the deprecated IgeMacIntegration which doesn't work
on 32-bit Macs, get the locale dir using the quartz_application_...
API from GtkOSXApplication.
---
 configure.ac                       | 2 +-
 gtksourceview/Makefile.am          | 4 ++--
 gtksourceview/gtksourceview-i18n.c | 8 +++-----
 3 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index ccfe986..42fe313 100644
--- a/configure.ac
+++ b/configure.ac
@@ -76,7 +76,7 @@ AM_CONDITIONAL(OS_OSX, test "$os_osx" = "yes")
 if test "$os_osx" = "yes"; then
 	AC_DEFINE([OS_OSX],[1],[Defined if os is Mac OSX])
 
-	PKG_CHECK_MODULES(IGE_MAC, ige-mac-integration)
+	PKG_CHECK_MODULES(GTK_MAC, gtk-mac-integration)
 fi
 
 # Check for Glade3
diff --git a/gtksourceview/Makefile.am b/gtksourceview/Makefile.am
index f3781d5..d1136a4 100644
--- a/gtksourceview/Makefile.am
+++ b/gtksourceview/Makefile.am
@@ -10,7 +10,7 @@ INCLUDES = 				\
 	-I$(top_srcdir) -I$(srcdir) 	\
 	$(DISABLE_DEPRECATED)		\
 	$(WARN_CFLAGS) 			\
-	$(IGE_MAC_CFLAGS)		\
+	$(GTK_MAC_CFLAGS)		\
 	$(DEP_CFLAGS)
 
 BUILT_SOURCES = 			\
@@ -96,7 +96,7 @@ nodist_libgtksourceview_2_0_la_SOURCES =\
 completion_providers = 							\
 	completion-providers/words/libgtksourcecompletionwords.la
 
-libgtksourceview_2_0_la_LIBADD = $(DEP_LIBS) $(IGE_MAC_LIBS) $(completion_providers)
+libgtksourceview_2_0_la_LIBADD = $(DEP_LIBS) $(GTK_MAC_LIBS) $(completion_providers)
 libgtksourceview_2_0_la_LDFLAGS = -no-undefined -export-symbols-regex "^gtk_source_.*"
 libgtksourceview_2_0_includedir = $(includedir)/gtksourceview-2.0/gtksourceview
 
diff --git a/gtksourceview/gtksourceview-i18n.c b/gtksourceview/gtksourceview-i18n.c
index e4db3eb..32f8068 100644
--- a/gtksourceview/gtksourceview-i18n.c
+++ b/gtksourceview/gtksourceview-i18n.c
@@ -24,7 +24,7 @@
 #endif
 
 #ifdef OS_OSX
-#include <ige-mac-bundle.h>
+#include <gtkosxapplication.h>
 #endif
 
 #include <string.h>
@@ -45,11 +45,9 @@ get_locale_dir (void)
 
 	g_free (win32_dir);
 #elif defined (OS_OSX)
-	IgeMacBundle *bundle = ige_mac_bundle_get_default ();
-
-	if (ige_mac_bundle_get_is_app_bundle (bundle))
+	if (quartz_application_get_bundle_id () != NULL)
 	{
-		locale_dir = g_strdup (ige_mac_bundle_get_localedir (bundle));
+		locale_dir = g_build_filename (quartz_application_get_resource_path (), "share", "locale", NULL);
 	}
 	else
 	{
-- 
2.8.0

